smac_planner/src/a_star.cpp

```diff
@@ -313,6 +321,32 @@ bool AStarAlgorithm<NodeT>::createPath(
     // 2) Mark Nbest as visited
     current_node->visited();
 
+    if (_motion_model == MotionModel::DUBIN || _motion_model == MotionModel::REEDS_SHEPP) {
```

Comment:
Place that logic into a new inline function `shouldAnalyticExpand()` or something.

Reply:
I think this was more important when we wanted to avoid having 2 node template overwrites. Now that the NodeT that's non-SE2 just returns a nullptr, this check is unncessary now

Reply:
All the logic for the expansion is now in its own function. Much cleaner.